string.c

```diff
@@ -150,6 +150,12 @@ str_enc_fastpath(VALUE str)
     }
 }
 
+static inline bool
+str_enc_single_byte_fastpath(VALUE str)
```

Comment:
This looks very similar to `single_byte_optimizable`, which already fast-paths `ENC_CODERANGE_7BIT`. Would adding a fast-path check for `ENCINDEX_ASCII_8BIT` there be sufficient? That could have wide reaching positive impact.

Reply:
I like that idea. It means that on a miss we fetch the encoding twice, but that's probably OK.